### PR TITLE
Update ITemplateLinkingFunction signature according to ng1.6

### PR DESCRIPTION
**Pull request by kevinmerckx**
update ITemplateLinkingFunction according to latest angular version 1.6 and as described in https://docs.angularjs.org/api/ng/service/$compile

### CODE REVIEW

**Comment by andy-ms** [comment]
This adds a file `angularjs/angular.d.ts`, could you remove it?

**Comment by kevinmerckx** [comment]
@andy-ms done!

File: angular/index.d.ts [hunk 1]
```diff
@@ -1280,7 +1280,15 @@ declare namespace angular {
...
     // This corresponds to the "publicLinkFn" returned by $compile.
     interface ITemplateLinkingFunction {
-        (scope: IScope, cloneAttachFn?: ICloneAttachFunction): JQuery;
+        (scope: IScope, cloneAttachFn?: ICloneAttachFunction, options?: ITemplateLinkingFunctionOptions): JQuery;
+    }
+
+    interface ITemplateLinkingFunctionOptions {
+        parentBoundTranscludeFn?: ITranscludeFunction,
+        transcludeControllers?: {
+            [controller: string]: { instance: any }
```
Comment (thorn0) [inline angular/index.d.ts]: Let's replace `any` with `IController`.

Comment (kevinmerckx) [inline angular/index.d.ts]: you're right, on it

**Comment by thorn0** [comment]
:+1:

**Pull request merged**